### ISSUE TITLE
CoastGuard-BuoyDrop - Science Buoy Transmit not Completing Contract when RemoteTech installed

### ISSUE THREAD

**Issue by inigma2** [inigmatus/GAP#161]
Reported by Ringkeeper:
http://forum.kerbalspaceprogram.com/index.php?/topic/129208-contract-pack-giving-aircraft-a-purpose-gap-1021-air-flights-tours-coast-guard-sts-jan-11-2016/&do=findComment&comment=2380507

**Comment by inigma2** [comment]
from Ringkeeper:
The buoy was connected via a Sat to the Center, so that was not the problem. It uploaded also the science 100% (at least according to the status text in the top left corner) . But the contract did not recognise it.

Thats exactly the same problem  the survey scanner/science lab had until the community fix for them. So i guess its not a problem of the GAP contract but a bug in RT.

**Comment by inigma2** [comment]
Nightingale says if RemoteTech is throwing an exception on transmit, then it is a RemoteTech problem.

**Comment by inigma2** [comment]
Will need to test buoy transmit with RT installed.

**Comment by inigma2** [comment]
while testing, realized that transmitter check is failing due to RT rewriting the partModule in Module Manager. So fix is to add a check for RT and change

partModule = ModuleDataTransmitter          

to

partModule = ModuleRTAntenna

and with this change I as able to transmit just fine and get contract to complete through a relay satellite.

**Comment by inigma2** [comment]
Ran into problem implementing this. CKAN does not remove RemoteTech folder thus causing a patch based on NEEDS[RemoteTech] still to apply, and failing contract load since patched part does not exist. Currently not working:

@CONTRACT_TYPE[CoastGuard-DropBuoys]:FOR[KSC?Coast?Guard]:NEEDS[RemoteTech]:HAS[@MODULE[ModuleRTAntenna]]

so I gave up. and removed the antenna requirement. players should be able to figure out for themselves that their buoys need antennas based on the intro contract text, hopefully.
